### PR TITLE
Kitchen expansion part 2: Ovens

### PR DESCRIPTION
**Pull request by ghost**
:cl: AMonkeyThatCodes, Kryson
content: Adds an oven to the kitchen, which you can use to bake pizzas, pastries and other dough products.
/:cl:

This PR adds an oven to every kitchen which bakes dough products similarly to a griddle. Simply open it up, fill an oven tray with some cookies, toss it in, close it up, and wait!

The oven will prepare the cookies and when you see brown smoke (which i need to add still) the food is done. When it goes black you're a bit too late!

just like griddles, I think ovens will make cooking a lot more fun and intuitive compared to the previous "first you bake the dough then you throw cooked ingredients on top.

https://user-images.githubusercontent.com/20987591/127382777-c7eefe56-a31f-440b-b802-ddbee744e07e.mp4

todo:
- [x] Port all dough-based food to use this.
- [x] beg for raw food sprites
- [x] add smoke sprites (White for baking, brown for done, black for burnt)
- [x] add oven to all maps

### CODE REVIEW

**Comment by Time-Green** [comment]
Insanely cool and a long time coming! Now we just need a stone oven for pizza kitchen!

Where do you draw the line between oven and microwave? There’s some overlap (irl aswell) between the two. Perhaps make ovens also cook stuff like donkpockets, but just slower

**Comment by YakumoChen** [comment]
Make it so that if something burns for long enough it ~~creates an aurora borealis~~ starts a fire in the kitchen

**Comment by AMonkeyThatCodes** [comment]
@Time-Green the overlap is that anything that is raw cant be microwaved, making it so you can put donk pockets in the oven is a non-issue though and really easy to do.

**Comment by Time-Green** [comment]
> @Time-Green the overlap is that anything that is raw cant be microwaved, making it so you can put donk pockets in the oven is a non-issue though and really easy to do.

Ah that’s perfect 👍

**Comment by ilikesaying** [comment]
steamed hams

**Comment by AndroidMK2** [comment]
I gotta say, big poggers this is.

**Comment by maxymax13** [comment]
what if i were to disguise donk pockets as my own cooking

**Comment by AMonkeyThatCodes** [comment]
oven players when they see downstreams still using microwaves after this pr https://www.youtube.com/watch?v=Hc_xGSIUb90

File: code/__DEFINES/food.dm [hunk 1]
```diff
@@ -87,3 +87,16 @@
 #define ICE_CREAM_BLAND "bland"
 
 #define DEFAULT_MAX_ICE_CREAM_SCOOPS 3
+
+
+
+#define OVEN_SMOKE_STATE_NONE 0
```
Comment (AnturK) [inline code/__DEFINES/food.dm:93]: Is there any reason for these to be defined outside of oven file.

File: code/game/objects/items/food/_food.dm [hunk 2]
```diff
@@ -26,14 +26,16 @@
...
-	///What you get if you microwave the food, this should be replaced once I fully re-work cooking.
+	///What you get if you microwave the food. Use baking for raw things, use microwaving for already cooked things
 	var/microwaved_type
 	///Type of atom thats spawned after eating this item
 	var/trash_type
 	///How much junkiness this food has? God I should remove junkiness soon
 	var/junkiness
 	///Will this food turn into badrecipe on a grill? Don't use this for everything; preferably mostly for food that is made on a grill to begin with so it burns after some time
 	var/burns_on_grill = FALSE
+	///Will this food turn into badrecipe in an oven? Don't use this for everything; preferably mostly for food that is made in an oven to begin with so it burns after some time
```
Comment (AnturK) [inline code/game/objects/items/food/_food.dm:37]: If that's the purpose wouldn't it be better for this to check if it's bakeable with positive result then add the negative one automatically ?

File: code/__DEFINES/food.dm (hunk 1 above)
Comment (AMonkeyThatCodes) [reply code/__DEFINES/food.dm:93]: guess not

File: code/game/objects/items/food/meat.dm [hunk 3]
```diff
@@ -915,7 +915,6 @@
 	name = "raw crab meat"
 	desc = "A pile of raw crab meat."
 	icon_state = "crabmeatraw"
-	microwaved_type = /obj/item/food/meat/crab
```
Comment (dragomagol) [inline code/game/objects/items/food/meat.dm:918]: Is there another way to cook crab meat?

**Pull request merged**